src/EnergyPlus/DataZoneEquipment.hh

```diff
@@ -76,57 +76,73 @@ namespace DataZoneEquipment {
     // Thus, all variables in this module must be PUBLIC.
 
     // MODULE PARAMETER DEFINITIONS:
-    constexpr int PathInlet(1);
-    constexpr int CompInlet(2);
-    constexpr int Intermediate(3);
-    constexpr int Outlet(4);
+    enum class AirNodeType
+    {
+        Unassigned = -1,
+        PathInlet,
+        CompInlet,
+        Intermediate,
+        Outlet
+    };
 
-    constexpr int ZoneSplitter_Type(1);
-    constexpr int ZoneSupplyPlenum_Type(2);
-    constexpr int ZoneMixer_Type(3);
-    constexpr int ZoneReturnPlenum_Type(4);
+    enum class CompType
+    {
+        Unassigned = -1,
+        ZoneSplitter,
+        ZoneSupplyPlenum,
+        ZoneMixer,
+        ZoneReturnPlenum
+    };
 
     // Start zone equip objects
     // list units that are valid for zone system availability managers first
-    constexpr int FanCoil4Pipe_Num(1);
-    constexpr int PkgTermHPAirToAir_Num(2);
-    constexpr int PkgTermACAirToAir_Num(3);
-    constexpr int PkgTermHPWaterToAir_Num(4);
-    constexpr int WindowAC_Num(5);
-    constexpr int UnitHeater_Num(6);
-    constexpr int UnitVentilator_Num(7);
-    constexpr int ERVStandAlone_Num(8);
-    constexpr int VentilatedSlab_Num(9);
-    constexpr int OutdoorAirUnit_Num(10);
-    constexpr int VRFTerminalUnit_Num(11);
-    constexpr int PurchasedAir_Num(12);
-    constexpr int ZoneEvaporativeCoolerUnit_Num(13);
-    constexpr int ZoneHybridEvaporativeCooler_Num(14); // #14, last zone equipment type to use zone availability manager. The above list must not
-                                                       // change or NumValidSysAvailZoneComponents(14) must also change.
-    constexpr int AirDistUnit_Num(15);
-    constexpr int BBWaterConvective_Num(16);
-    constexpr int BBElectricConvective_Num(17);
-    constexpr int HiTempRadiant_Num(18);
-    constexpr int LoTempRadiant_Num(19);
-    constexpr int ZoneExhaustFan_Num(20);
-    constexpr int HeatXchngr_Num(21);
-    constexpr int HPWaterHeater_Num(22);
-    constexpr int BBWater_Num(23);
-    constexpr int ZoneDXDehumidifier_Num(24);
-    constexpr int BBSteam_Num(25);
-    constexpr int BBElectric_Num(26);
-    constexpr int RefrigerationAirChillerSet_Num(27);
-    constexpr int UserDefinedZoneHVACForcedAir_Num(28);
-    constexpr int CoolingPanel_Num(29);
-    constexpr int ZoneUnitarySys_Num(30);
-
-    // Per Person Ventilation Rate Mode
-    constexpr int PerPersonDCVByCurrentLevel(1);
-    constexpr int PerPersonByDesignLevel(2);
+    enum ZoneEquip
+    {
+        Unassigned = -1,
+        FanCoil4Pipe = 1,
```

Comment:
Why is this not 0?

Reply:
@dareumnam any comment on this?

Reply:
![image](https://user-images.githubusercontent.com/5778156/142266425-f67650a3-705b-4c28-88ab-8a0fcc861641.png)

Reply:
Per my comment above, arrays indexed by `enum` should be turned into `std::array`. Per same comment, no need to hold up this PR for that single change.

Reply:
:+1: